src/components/SwitchErrorBoundary/SwitchErrorBoundary.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { Route, Switch } from 'react-router-dom';
+
+type ErrorHandlerFunction = (error: Error, componentStack: string) => void;
+
+type SwitchErrorBoundaryProps = {
+  fallBackComponent: any;
+  onError?: ErrorHandlerFunction;
+};
+
+type SwitchErrorBoundaryState = {
+  hasError: boolean;
+};
+
+export default class SwitchErrorBoundary extends React.Component<SwitchErrorBoundaryProps, SwitchErrorBoundaryState> {
+  private show: boolean;
+
+  constructor(props: SwitchErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false };
+    this.show = false;
+  }
+
+  componentDidCatch(error: any, info: any) {
+    if (this.props.onError) {
+      this.props.onError(error, info);
+    }
+    this.setState({ hasError: true });
```

Comment:
This is great for now, but will be cool to see this integrated into redux.

Reply:
Yes! that's the idea :)